geotiepoints/modisinterpolator.py

```diff
@@ -84,177 +85,214 @@ def compute_expansion_alignment(satz_a, satz_b, satz_c, satz_d, scan_width):
-            self.fscan_full_width = 1354 * 2
-            self.fscan_len = 2 * 10 // self.cscan_len
-        elif fres == 1000:
-            self.fscan_width = 1 * self.cscan_width
-            self.fscan_full_width = 1354
-            self.fscan_len = 1 * 10 // self.cscan_len
-
-    def _expand_tiepoint_array_1km(self, arr, lines, cols):
-        arr = da.repeat(arr, lines, axis=1)
-        arr = da.concatenate((arr[:, :lines//2, :], arr, arr[:, -(lines//2):, :]), axis=1)
-        arr = da.repeat(arr.reshape((-1, self.cscan_full_width - 1)), cols, axis=1)
-        return da.hstack((arr, arr[:, -cols:]))
+@scanline_mapblocks
+def _interpolate(
+        lon1,
+        lat1,
+        satz1,
+        coarse_resolution=None,
+        fine_resolution=None,
+        coarse_scan_width=None,
+):
+    """Helper function to interpolate scan-aligned arrays.
 
-    def _get_coords_1km(self, scans):
-        y = (np.arange((self.cscan_len + 1) * self.fscan_len) % self.fscan_len) + .5
-        y = y[self.fscan_len // 2:-(self.fscan_len // 2)]
-        y[:self.fscan_len//2] = np.arange(-self.fscan_len/2 + .5, 0)
-        y[-(self.fscan_len//2):] = np.arange(self.fscan_len + .5, self.fscan_len * 3 / 2)
-        y = np.tile(y, scans)
+    This function's decorator runs this function for each dask block/chunk of
+    scans. The arrays are scan-aligned meaning they are an even number of scans
+    (N rows per scan) and contain the entire scan width.
 
-        x = np.arange(self.fscan_full_width) % self.fscan_width
-        x[-self.fscan_width:] = np.arange(self.fscan_width, self.fscan_width * 2)
-        return x, y
+    """
+    interp = _Interpolator(coarse_resolution, fine_resolution, coarse_scan_width=coarse_scan_width)
+    return interp.interpolate(lon1, lat1, satz1)
 
-    def _expand_tiepoint_array_5km(self, arr, lines, cols):
-        arr = da.repeat(arr, lines * 2, axis=1)
-        arr = da.repeat(arr.reshape((-1, self.cscan_full_width - 1)), cols, axis=1)
-        factor = self.fscan_width // self.cscan_width
-        if self.cscan_full_width == 271:
-            return da.hstack((arr[:, :2 * factor], arr, arr[:, -2 * factor:]))
-        else:
-            return da.hstack((arr[:, :2 * factor], arr, arr[:, -self.fscan_width:], arr[:, -2 * factor:]))
 
-    def _get_coords_5km(self, scans):
-        y = np.arange(self.fscan_len * self.cscan_len) - 2
-        y = np.tile(y, scans)
-
-        x = (np.arange(self.fscan_full_width) - 2) % self.fscan_width
-        x[0] = -2
-        x[1] = -1
-        if self.cscan_full_width == 271:
-            x[-2] = 5
-            x[-1] = 6
-        elif self.cscan_full_width == 270:
-            x[-7] = 5
-            x[-6] = 6
-            x[-5] = 7
-            x[-4] = 8
-            x[-3] = 9
-            x[-2] = 10
-            x[-1] = 11
-        else:
-            raise NotImplementedError("Can't interpolate if 5km tiepoints have less than 270 columns.")
-        return x, y
+class _Interpolator:
```

Comment:
Even if this is private, I think it should have Modis in its name since it is specific to that instrument.

Reply:
I guess...but it is in the `modisinterpolator.py` module. If I change that would you be OK with a merge without another review?

Reply:
yes!